docs/DeveloperGuide.md

```diff
@@ -1,18 +1,133 @@
 
+### Architecture
+
+![ArchitectureDiagram](/docs/images/Architecture.png)
+
+The Architecture Diagram given above explains the high-level design of the Diet Manager Application.
+
+The components involved are given below:
+
+`UI`: The user interface of the application
+
+`Storage`: Reads data from and writes data to the specified data files
+
+`Parser`: Reads the given user input and breaks it down into machine-readable code
+
+`Profile`: Stores and records the user's profile information
+
+### Logic component
+
+The `Logic` component is responsible for:
+* Arranging the whole workflow
+* Deciding how functional components interact with each other. 
+
+The Logic consists of the following class:
+* DietManager-Arranges the main workflow of the program.
+
+### UI component
+
+The `UI` component is responsible for:
+* Receiving all command line inputs from the user
+* Displaying all system outputs to the user
+
+The UI consists of the following classes: 
+* `UI` - Reads user input and displays system output
+* `MessageBank` - Stores all standard system output messages
+
+### Storage component
+
+The `Storage` component is responsible for:
+* Managing the respective data files
+* Storing all system logs generated from the application
+
+The Storage consists of the following classes: 
+* `Storage` - Stores all user profile information in respective data files
+* `AppLogger` - Stores all system logs generated from the application
+
+
+### Parser component
+
+The `Parser` component is responsible for:
+* Reading user input and breaking it down to machine-readable code
+
+The Parser consists of the following classes: 
+* `Parser` - Breaks down inputs into machine-readable code
+
+### Profile component
+
+The `Profile` component is responsible for:
+* Storing all user profile information
+
+The Profile consists of the following classes: 
+* `Profile` - 
+* `DailyFoodRecord` - 
+
+### Food component
+
+The `Food` component is responsible for:
+* 
+
+The Food consists of the following classes: 
+* `Food` - 
+* `FoodNutritionInfo` - 
+
+
+## Implementation
+
+{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### 1. [Proposed] Record Meal Feature
+#### 1.1 Proposed implementation
+The record feature is facilitated by `RecordMealCommand`. It extends `Command` and overrides `execute()` and `saveResults()`
+
+Given below is an example usage scenario and how the record mechanism behaves at each step.
+
+Step 1.  The `ui` object gets user input and sends it to the `parser`. The `parser` then parses the original input into a standard `command`
+and returns it. The command type is decided by the `commandPrompt`. The `RecordMealCommand` has `record-meal` as its commandPrompt. 
+During the process a new`RecordMealCommand` object will be generated and returned to the `Logic` component.
+
+### Step1. Generate command
+![Step1. Diagram](/docs/images/Record_step1.png)
```

Comment:
In lines 38 and 39 of this document, the `Logic` component consists of `DietManager-Arranges`. Perhaps it would be clearer if the diagram's object would be using the class you have come up with, `:Arranges`, instead of `:Logic`.

Here's a sample that you can consider:
![image](https://user-images.githubusercontent.com/32288711/77891100-0d80c500-72a3-11ea-991d-163b02bcfc3a.png)